presto-main/src/main/java/com/facebook/presto/memory/LocalMemoryManager.java

```diff
@@ -42,16 +45,43 @@
     public LocalMemoryManager(NodeMemoryConfig config)
     {
         requireNonNull(config, "config is null");
-        maxMemory = new DataSize(Runtime.getRuntime().maxMemory(), BYTE);
-        checkArgument(config.getMaxQueryMemoryPerNode().toBytes() <= maxMemory.toBytes(), format("%s set to %s, but only %s of useable heap available", QUERY_MAX_MEMORY_PER_NODE_CONFIG, config.getMaxQueryMemoryPerNode(), maxMemory));
+        long availableMemory = Runtime.getRuntime().maxMemory();
+        validateHeapHeadroom(config, availableMemory);
+
+        maxMemory = new DataSize(availableMemory - config.getHeapHeadroom().toBytes(), BYTE);
+        validateConfiguredMaxMemory(config.getMaxQueryMemoryPerNode().toBytes(), QUERY_MAX_MEMORY_PER_NODE_CONFIG);
+        validateConfiguredMaxMemory(config.getMaxQueryTotalMemoryPerNode().toBytes(), QUERY_MAX_TOTAL_MEMORY_PER_NODE_CONFIG);
```

Comment:
If I understand correctly. The second `validateConfiguredMaxMemory` does nothing. The first `validateConfiguredMaxMemory` is basically asserting that `MaxQueryMemoryPerNode < MaxQueryTotalMemoryPerNode`.

Remove the second, and rewrite the first one?